bundles/mapping/mapmodule/service/map.layer.js

```diff
@@ -329,10 +344,191 @@ Oskari.clazz.define('Oskari.mapframework.service.MapLayerService',
                 this._populateWmsMapLayerAdditionalData(layer, newLayerConf);
             }
 
+            // Also update layer to me._layerGroups
+            this.updateGroupsLayers(layerId, newLayerConf);
+
             // notify components of layer update
             var evt = Oskari.eventBuilder('MapLayerEvent')(layer.getId(), 'update');
             this._sandbox.notifyAll(evt);
         },
+        /**
+         * Delete layer group
+         * @method deleteLayerGroup
+         * @param  {Integer}         id group id
+         * @param {Integer}          parentId parent id
+         */
+        deleteLayerGroup: function(id, parentId) {
+            var me = this;
+            var newGroups = [];
+            var editable = me.getAllLayerGroups(parentId);
+            editable.groups.forEach(function(group) {
+                if (group.id != id) {
+                    newGroups.push(group);
+                }
+            });
+            editable.groups = newGroups;
+        },
+        /**
+         * Updata layer groups
+         * @method updateLayerGroups
+         * @param  {Object}          data data
+         */
+        updateLayerGroups: function(data) {
+            var me = this;
+            var editable = me.getAllLayerGroups(data.id);
+            // if found then update only
+            if (editable) {
+                editable.name = data.name;
+                editable.parentId = data.parentId;
+                editable.selectable = data.selectable;
+            } else if (data.parentId === -1) {
+                me.getAllLayerGroups().push({
+                    groups: [],
+                    id: data.id,
+                    name: data.name,
+                    layers: [],
+                    parentId: data.parentId,
+                    selectable: data.selectable
+                });
+            } else {
+                me.getAllLayerGroups(data.parentId).groups.push({
+                    groups: [],
+                    id: data.id,
+                    name: data.name,
+                    layers: [],
+                    parentId: data.parentId,
+                    selectable: data.selectable
+                });
+            }
+        },
+
+
+        /**
+         * Update layer groups
+         * @method updateGroupsLayers
+         * @param  {String}          layerId      layerid
+         * @param  {Object}          newLayerConf new layer conf
+         * @param  {Boolean}         deleteLayer delete layer
+         * @param  {Boolean}         newLayer is new layer
+         */
+        updateGroupsLayers: function(layerId, newLayerConf, deleteLayer, newLayer) {
```

Comment:
There's a function updateLayerGroups and updateGroupsLayers. With comments "Updata layer groups" and "Update layer groups". Please check the comments so they are relevant and try to find a better name for updateGroupsLayers.

Reply:
Is updateGroupsLayers updating the layers inside groups? If so maybe it could be updateLayersInGroups?

Reply:
Why is the newLayerConf passed here and not the instance of AbstractLayer?